drivers/char/vhm/vhm_dev.c

```diff
@@ -395,6 +395,23 @@ static long vhm_dev_ioctl(struct file *filep,
 		break;
 	}
 
+	case IC_CLEAR_VM_IOREQ: {
+		/*
+		 * TODO: Query VM status with additional hypercall.
+		 * VM should be in paused status.
+		 *
+		 * In SMP SOS, we need flush the current pending ioreq dispatch
+		 * tasklet and finish it before clearing all ioreq of this VM.
+		 * With tasklet_kill, there still be a very rare race which
+		 * might lost one ioreq tasklet for other VMs. So arm one after
+		 * the clearing. It's harmless.
+		 */
+		tasklet_kill(&vhm_io_req_tasklet);
```

Comment:
Can we  call the tasklet_schedule firstly and then tasklet_kill so that the kernel can distribute the io_request? (The tasklet_kill is used to assure that the distribution of io_reuqest is completed).

After it is done, we can clear the corresponding io_request safely for the given vm.

Reply:
Not need. tasklet_kill can guarantee the tasklet flushing.

Reply:
Talked with yakui, If IPI blocked by one core of SOS and ioctl running on another core of SOS,  we might lost one ioreq of current VM. I add one more tasklet_schedule before tasklet_kill.